### ISSUE TITLE
Sticky element disappears at bottom of parent div

### ISSUE THREAD

**Issue by Globegitter** [leafo/sticky-kit#31]
Tried out your plugin and for most part it is working better than any other of the sticky plugins that I tried, so thanks for the great work on that side. But there is one bug. I have this video-element which should stay sticky and it does, but when I scroll to the very bottom of the page it just disappears like this:

![capture decran 2014-03-04 a 15 37 01](https://f.cloud.github.com/assets/2580254/2334517/92b736ae-a479-11e3-9421-8a20c2ceb6f4.png)

Instead of looking like 'sticking' to the end of the parent div like this (this is using `position: sticky` in the firefox nightly)
![capture decran 2014-03-04 a 15 36 53](https://f.cloud.github.com/assets/2580254/2334526/a946085a-a479-11e3-909c-1778d6ce32c7.png)

**Comment by abrahamian** [comment]
Thank you, @danxshap , for that fix! A tip to other users: for styling purposes it is useful to add a class to the 'parent' div into which sticky-kit places the target element. 

I personally found I had better control when removing the mother div's inline style attribute altogether, and adding my own css rules. My code (using angular.js and sass)  includes:

js:

	element.on('sticky_kit:stick', function(e) {
      element.parent().removeAttr('style');
      element.parent().addClass('mother-of-sticky');
    })
    .on('sticky_kit:bottom', function(e){
      element.parent().addClass('bottomed');
    })
    .on('sticky_kit:unbottom', function(e){
      element.parent().removeClass('bottomed')
    });

scss:

  	.mother-of-sticky {
    	@extend .col-sm-5;
	    @extend .no-padding;
  
    	position:relative;
    
	    &.bottomed {
    	  position: static;
	    }
  
    	.is_stuck {
      		padding-right:15.5px;  
	    }
	  }

**Comment by hh2k** [comment]
I got it fixed by placing the code in a $(window).load(function(){ ... }); function.

**Comment by zeljkoprsa** [comment]
@hh2k That worked for me too.

**Comment by JustMikey** [comment]
I've tried all the methods above and tried to understand why it worked incorrectly and I failed, then I did what @hh2k said, and well, it worked -_- So if you're having the same problem, try his method first and if it doesnt help, then try any from the above

**Comment by nlifors** [comment]
None of these suggestions worked for me unfortunately, but I found another way to fix this, based on  Globegitter's comment above. I had to set the css 'position', 'top', and 'bottom', as these are set by the plugin as soon as the element is 'stuck'.

Assuming sticky element with an id of 'pinned'. Also, the top:30px is arbitrary, and should be set to offset_top option value. This can be replaced with 0px instead if no space is needed.

```javascript
$('#pinned').on('sticky_kit:bottom', function(e) {
         $(this).parent().css('position', 'static');
         $(this).css({
            'position': 'fixed',
            'top': '50px',
            'bottom': ""
         });
}).on('sticky_kit:unbottom', function(e) {
         $(this).parent().css('position', 'relative');
})
```

**Comment by jelen07** [comment]
@danxshap Thanks for your workaround. I am using TB3 with sticky-kit and I've had the same problem. Now everything works fine. :+1:

**Comment by P4KM4N** [comment]
Thanks @danxshap That was exactly what I needed! It should be in the docs!

**Comment by PrincessRebaula** [comment]
hi, does anybody encountered a problem with the sticky kit in safari?
it's not working in safari browser. I'm using the sticky kit for my sidebar

this the javascript which I saved as another js file and include it in my main page as well as the other scripts and css needed 

(function() {
  var reset_scroll;

 $(document).ready( function() {
    return $("[data-sticky_column]").stick_in_parent({
      parent: "[data-sticky_parent]"
    });
  });

  reset_scroll = function() {
    var scroller;
    scroller = $("body,html");
    scroller.stop(true);
    if ($(window).scrollTop() !== 0) {
      scroller.animate({
        scrollTop: 0
      }, "fast");
    }
    return scroller;
  };

  window.scroll_it = function() {
    var max;
    max = $(document).height() - $(window).height();
    return reset_scroll().animate({
      scrollTop: max
    }, max * 3).delay(100).animate({
      scrollTop: 0
    }, max * 3);
  };

  window.scroll_it_wobble = function() {
    var max, third;
    max = $(document).height() - $(window).height();
    third = Math.floor(max / 3);
    return reset_scroll().animate({
      scrollTop: third * 2
    }, max * 3).delay(100).animate({
      scrollTop: third
    }, max * 3).delay(100).animate({
      scrollTop: max
    }, max * 3).delay(100).animate({
      scrollTop: 0
    }, max * 3);
  };

  $(window).on("resize", (function(_this) {
    return function(e) {
      return $(document.body).trigger("sticky_kit:recalc");
    };
  })(this));

}).call(this);

----
this is the html, by the way i'm using bootstrap also

<div class="container">
	<div class="row" data-sticky_parent="" style="position: relative;">
		<div class="row-same-height">
			<div class="col-md-3 col-md-height col-top ">
				<div data-sticky_column="">
                     .............some content here...............
                	         </div>
                          </div>

             <div class="col-md-9 col-md-height col-top ">
             		-----very long content here.......
             </div>
        </div>
     </div>
</div>


-thanks for help

**Comment by PrincessRebaula** [comment]
hi, does anybody encountered a problem with the sticky kit in safari?
it's not working in safari browser. I'm using the sticky kit for my sidebar

this the javascript which I saved as another js file and include it in my main page as well as the other scripts and css needed

(function() {
var reset_scroll;

$(document).ready( function() {
return $("[data-sticky_column]").stick_in_parent({
parent: "[data-sticky_parent]"
});
});

reset_scroll = function() {
var scroller;
scroller = $("body,html");
scroller.stop(true);
if ($(window).scrollTop() !== 0) {
scroller.animate({
scrollTop: 0
}, "fast");
}
return scroller;
};

window.scroll_it = function() {
var max;
max = $(document).height() - $(window).height();
return reset_scroll().animate({
scrollTop: max
}, max * 3).delay(100).animate({
scrollTop: 0
}, max * 3);
};

window.scroll_it_wobble = function() {
var max, third;
max = $(document).height() - $(window).height();
third = Math.floor(max / 3);
return reset_scroll().animate({
scrollTop: third * 2
}, max * 3).delay(100).animate({
scrollTop: third
}, max * 3).delay(100).animate({
scrollTop: max
}, max * 3).delay(100).animate({
scrollTop: 0
}, max * 3);
};

$(window).on("resize", (function(_this) {
return function(e) {
return $(document.body).trigger("sticky_kit:recalc");
};
})(this));

}).call(this);

this is the html, by the way i'm using bootstrap also

![capture](https://cloud.githubusercontent.com/assets/10446279/8640949/2fb27ff2-293c-11e5-9b28-84a9782dfcf3.JPG)

-thanks for help

**Comment by sefod** [comment]
One more thumb up for @danxshap solution!

**Comment by mebbopakko** [comment]
Yep, thanks @danxshap, worked like a charm :+1:

**Comment by jume-dev** [comment]
and another one, works perfectly :+1:

**Comment by Cacskate** [comment]
@danxshap is awesome!

**Comment by 6233114** [comment]
@danxshap love your work!

**Comment by geo1004** [comment]
Well... That's a fix ! Thanks @danxshap

**Comment by izn** [comment]
Well played, @danxshap :+1:
It's working!

**Comment by jothanluna** [comment]
@danxshap i have the same issue  but only with firefox, are you have an idea what is happen? http://front.delitogo.mx/restaurantes/13

**Comment by mimomedia** [comment]
Hi, i just solved the thing using the spacer feature(use own spacer) of sticky-kit and applying in my css, 'display:inline-block' to the spacer element, following the demo markup.

Regards to everyone

**Comment by chaegumi** [comment]
Here is my solution.

```html

<div class="container">
	<div class="row" data-sticky_parent>
		<div class="col-sm-9 col-sm-push-3" data-sticky_column>
		
		</div>
		<div id="left_menu" class="col-sm-3 col-sm-pull-9" data-sticky_column>
			
		</div>
	</div>
</div>

```

```javascript

$(function(){
	$("#left_menu").stick_in_parent({inner_scrolling:false,offset_top:70})
		.on('sticky_kit:stick', function(e) {
				$('#left_menu').parent().addClass('col-sm-3 col-sm-pull-9');
				$('#left_menu').removeClass('col-sm-3 col-sm-pull-9');
            }).on('sticky_kit:unstick', function(e) {
				$('#left_menu').addClass('col-sm-3 col-sm-pull-9');
			}).on('sticky_kit:bottom', function(e){	
				$('#left_menu').parent().removeClass('col-sm-3 col-sm-pull-9');
				$('#left_menu').addClass('col-sm-3 col-sm-pull-9');
				
				$('#left_menu').parent().css('position', 'static');
            }).on('sticky_kit:unbottom', function(e){
				$('#left_menu').parent().addClass('col-sm-3 col-sm-pull-9');
				$('#left_menu').removeClass('col-sm-3 col-sm-pull-9');
				
				$('#left_menu').parent().css('position', 'relative');
			});
	if(screen.width <= 768){
		$('#left_menu').trigger("sticky_kit:detach");
	}		
});


```

**Comment by dskvr** [comment]
I cannot believe this issue still exists :man_facepalming:

**Comment by kumarancontus** [comment]
Hi,

I am using the following jQuery http://leafo.net/sticky-kit/#reference for sidebar sticky but in my case I want to make the sidebar position as absolute when reaching a custom div instead of footer, how can I achieve this?

**My script**

(function(){var b,f;b=this.jQuery||window.jQuery;f=b(window);b.fn.stick_in_parent=function(d){var A,w,J,n,B,K,p,q,k,E,t;null==d&&(d={});t=d.sticky_class;B=d.inner_scrolling;E=d.recalc_every;k=d.parent;q=d.offset_top;p=d.spacer;w=d.bottoming;null==q&&(q=0);null==k&&(k=void 0);null==B&&(B=!0);null==t&&(t="is_stuck");A=b(document);null==w&&(w=!0);J=function(a,d,n,C,F,u,r,G){var v,H,m,D,I,c,g,x,y,z,h,l;if(!a.data("sticky_kit")){a.data("sticky_kit",!0);I=A.height();g=a.parent();null!=k&&(g=g.closest(k));
if(!g.length)throw"failed to find stick parent";v=m=!1;(h=null!=p?p&&a.closest(p):b("<div />"))&&h.css("position",a.css("position"));x=function(){var c,f,e;if(!G&&(I=A.height(),c=parseInt(g.css("border-top-width"),10),f=parseInt(g.css("padding-top"),10),d=parseInt(g.css("padding-bottom"),10),n=g.offset().top+c+f,C=g.height(),m&&(v=m=!1,null==p&&(a.insertAfter(h),h.detach()),a.css({position:"",top:"",width:"",bottom:""}).removeClass(t),e=!0),F=a.offset().top-(parseInt(a.css("margin-top"),10)||0)-q,
u=a.outerHeight(!0),r=a.css("float"),h&&h.css({width:a.outerWidth(!0),height:u,display:a.css("display"),"vertical-align":a.css("vertical-align"),"float":r}),e))return l()};x();if(u!==C)return D=void 0,c=q,z=E,l=function(){var b,l,e,k;if(!G&&(e=!1,null!=z&&(--z,0>=z&&(z=E,x(),e=!0)),e||A.height()===I||x(),e=f.scrollTop(),null!=D&&(l=e-D),D=e,m?(w&&(k=e+u+c>C+n,v&&!k&&(v=!1,a.css({position:"fixed",bottom:"",top:c}).trigger("sticky_kit:unbottom"))),e<F&&(m=!1,c=q,null==p&&("left"!==r&&"right"!==r||a.insertAfter(h),
h.detach()),b={position:"",width:"",top:""},a.css(b).removeClass(t).trigger("sticky_kit:unstick")),B&&(b=f.height(),u+q>b&&!v&&(c-=l,c=Math.max(b-u,c),c=Math.min(q,c),m&&a.css({top:c+"px"})))):e>F&&(m=!0,b={position:"fixed",top:c},b.width="border-box"===a.css("box-sizing")?a.outerWidth()+"px":a.width()+"px",a.css(b).addClass(t),null==p&&(a.after(h),"left"!==r&&"right"!==r||h.append(a)),a.trigger("sticky_kit:stick")),m&&w&&(null==k&&(k=e+u+c>C+n),!v&&k)))return v=!0,"static"===g.css("position")&&g.css({position:"relative"}),
a.css({position:"absolute",bottom:d,top:"auto"}).trigger("sticky_kit:bottom")},y=function(){x();return l()},H=function(){G=!0;f.off("touchmove",l);f.off("scroll",l);f.off("resize",y);b(document.body).off("sticky_kit:recalc",y);a.off("sticky_kit:detach",H);a.removeData("sticky_kit");a.css({position:"",bottom:"",top:"",width:""});g.position("position","");if(m)return null==p&&("left"!==r&&"right"!==r||a.insertAfter(h),h.remove()),a.removeClass(t)},f.on("touchmove",l),f.on("scroll",l),f.on("resize",
y),b(document.body).on("sticky_kit:recalc",y),a.on("sticky_kit:detach",H),setTimeout(l,0)}};n=0;for(K=this.length;n<K;n++)d=this[n],J(b(d));return this}}).call(this);

**Comment by yaster25** [comment]
@danxshap Thanks! $(document).ready helps

**Comment by misterserge** [comment]
@danxshap you're save my day! many thanks!

**Comment by catalin88** [comment]
This is 2019 and none of the above worked for me on woocommerce shop page :(

**Comment by wildtony** [comment]
@thejmill https://adroit6.github.io/pqDVTk82y3N4OBW1LreCtTGAdNG0aNKfL8fSNXUFZeiFK5HNRDwhjgI/
